converter/src/main/java/gov/cms/qpp/conversion/decode/SupplementalDataRaceDecoder.java

```diff
@@ -0,0 +1,49 @@
+package gov.cms.qpp.conversion.decode;
+
+import gov.cms.qpp.conversion.Context;
+import gov.cms.qpp.conversion.model.Decoder;
+import gov.cms.qpp.conversion.model.Node;
+import gov.cms.qpp.conversion.model.TemplateId;
+import gov.cms.qpp.conversion.model.validation.SupplementalData.SupplementalType;
+import java.util.function.Consumer;
+import org.jdom2.Attribute;
+import org.jdom2.Element;
+import org.jdom2.filter.Filters;
+
+/**
+ * Decoder for Supplemental Data Race Element
+ */
+@Decoder(TemplateId.RACE_SUPPLEMENTAL_DATA_ELEMENT_CMS_V2)
+public class SupplementalDataRaceDecoder extends QppXmlDecoder {
+
+	public SupplementalDataRaceDecoder(Context context) {
+		super(context);
+	}
+
+	/**
+	 *
+	 * @param element Top element in the XML document
+	 * @param thisNode Top node created in the XML document
+	 * @return
+	 */
+	@Override
+	protected DecodeResult internalDecode(Element element, Node thisNode) {
+		setRaceOnNode(element, thisNode);
+		return DecodeResult.TREE_CONTINUE;
+	}
+
+	/**
+	 * Sets the Race Code in the current Node
+	 *
+	 * @param element XML element that represents SupplementalDataCode
+	 * @param thisNode Current Node to decode into
+	 */
+	private void setRaceOnNode(Element element, Node thisNode) {
```

Comment:
This method is almost a duplicate of setPayerOnNode in SupplementalDataPayerDecoder

Reply:
I'll try to think of a good interface of sorts to remove it.